src/hooks/useNFTContract.js

```diff
@@ -0,0 +1,311 @@
+          //   callback(minttx);
+          callback();
+          const res = await minttx.wait();
+          if (res.transactionHash) {
+            afterminting();
+          }
+        } catch (err) {
+          console.warn('minting to webaverse contract failed');
+          setError('Mint Failed');
+        }
+      }
+      setMinting(false);
+    } catch (err) {
+      console.warn('minting to webaverse contract failed');
+      setError('Mint Failed');
+      setMinting(false);
+    }
+  }
+
+  async function mintfromVoucher(app, previewImage = 'testimageurl', callback = () => {}) {
+    setMinting(true);
+    setError('');
+    try {
+      let imageURI;
+      let avatarURI;
+
+      const signer = await getSigner();
+      const name = app.name;
+      const ext = 'testType';
+      //   const hash = currentApp.contentId.split(FILE_ADDRESS)[1].split('/')[0];
+      const description = 'testDescription';
+
+      const metadataFileName = `${name}-metadata.json`;
+      let metadata;
+      if (previewImage) { // 3D object
+        imageURI = previewImage;
+        avatarURI = 'testcontentID';
+
+        metadata = {
+          name,
+          description,
+          image: imageURI,
+          animation_url: avatarURI,
+        };
+      } else { // image object
+        imageURI = 'testcontentID';
+        avatarURI = '';
+
+        metadata = {
+          name,
+          description,
+          image: imageURI,
+        };
+      }
+
+      const type = 'upload';
+      let load = null;
+      // const json_hash = await handleBlobUpload(metadataFileName, JSON.stringify(metadata) )
+      // handleBlobUpload
+      // new Blob([JSON.stringify(metadata)], {type: 'text/plain'});
+      const json_hash = await handleBlobUpload(metadataFileName, new Blob([JSON.stringify(metadata)], {type: 'text/plain'}), {
+        onTotal(total) {
+          load = registerLoad(type, metadataFileName, 0, total);
+        },
+        onProgress(e) {
+          if (load) {
+            load.update(e.loaded, e.total);
+          } else {
+            load = registerLoad(type, metadataFileName, e.loaded, e.total);
+          }
+        },
+      });
+      if (load) {
+        load.end();
+      }
+
+      const metadataHash = json_hash.split(FILE_ADDRESS)[1].split('/')[0];
+      const webaverseContract = new ethers.Contract(WebaversecontractAddress, WebaverseABI, signer);
+      // const NFTcontract = new ethers.Contract(NFTcontractAddress, NFTABI, signer);
+      const FTcontract = new ethers.Contract(FTcontractAddress, FTABI, signer);
+
+      const bigMintFee = await webaverseContract.mintFee();
+      const mintfee = BigNumber.from(bigMintFee).toNumber();
+      if (mintfee > 0) { // webaverse side chain mintfee != 0
+        const FTapprovetx = await FTcontract.approve(NFTcontractAddress, mintfee); // mintfee = 10 default
+        const FTapproveres = await FTapprovetx.wait();
```

Comment:
`FTapprovers`?

Reply:
FTapproveres is for checking transactionHash.
if(FTapproveres.transactionHash)